TerrorConsole/Assets/_Main/Scripts/Objects/Switchs/SwitchLever.cs

```diff
@@ -0,0 +1,37 @@
+using UnityEngine;
+
+namespace TerrorConsole
+{
+    public class SwitchLever : SwitchObject
+    {
+        private bool _isPlayerColliding;
+        private void OnCollisionEnter2D(Collision2D collision)
+        {
+            if (collision.gameObject.CompareTag("Player"))
+            {
+                _isPlayerColliding = true;
+            }
+        }
+
+        private void OnCollisionExit2D(Collision2D collision)
+        {
+            if (collision.gameObject.CompareTag("Player"))
+            {
+                _isPlayerColliding = false;
+            }
+        }
+
+        private void Update()
+        {
+            if (!_isPlayerColliding)
+            {
+                return;
+            }
+
+            if (InputManager.Source.ActionButton1)
+            {
+                AlternateState();
+            }
+        }
+    }
```

Comment:
This will need to be changed to subscribe the button event instead of checking on update.